### ISSUE TITLE
atexit hooks are wrongly used

### ISSUE THREAD

**Issue by GoogleCodeExporter** [agronholm/pythonfutures#33]
```
I do not know how to reproduce with small program, but when using tornado 
tests, I receive that:

--------------------------------
Error in atexit._run_exitfuncs:
Traceback (most recent call last):
  File "/usr/lib/python2.7/atexit.py", line 24, in _run_exitfuncs
    func(*targs, **kargs)
  File "/home/mmarkk/.local/lib/python2.7/site-packages/concurrent/futures/process.py", line 82, in _python_exit
    items = list(_threads_queues.items())
AttributeError: 'NoneType' object has no attribute 'items'
Error in atexit._run_exitfuncs:
Traceback (most recent call last):
  File "/usr/lib/python2.7/atexit.py", line 24, in _run_exitfuncs
    func(*targs, **kargs)
  File "/home/mmarkk/.local/lib/python2.7/site-packages/concurrent/futures/thread.py", line 41, in _python_exit
    items = list(_threads_queues.items())
AttributeError: 'NoneType' object has no attribute 'items'
Error in sys.exitfunc:
Traceback (most recent call last):
  File "/usr/lib/python2.7/atexit.py", line 24, in _run_exitfuncs
    func(*targs, **kargs)
  File "/home/mmarkk/.local/lib/python2.7/site-packages/concurrent/futures/thread.py", line 41, in _python_exit
    items = list(_threads_queues.items())
AttributeError: 'NoneType' object has no attribute 'items'
---------------------------

But this problem is easily fixed with that:

--- process.py	2015-01-21 15:27:49.458086156 +0500
+++ process.py.new	2015-01-21 15:28:10.906259116 +0500
@@ -76,7 +76,7 @@
 _threads_queues = weakref.WeakKeyDictionary()
 _shutdown = False
 
-def _python_exit():
+def _python_exit(_threads_queues=_threads_queues):
     global _shutdown
     _shutdown = True
     items = list(_threads_queues.items())


!!!!! and exactly the same for threads.py !!!!

What the hell? I do not know how the _threads_queues may become None.

```

Original issue reported on code.google.com by `socketp...@gmail.com` on 21 Jan 2015 at 10:29

**Comment by agronholm** [comment]
Need code to reproduce this.

**Comment by hachreak** [comment]
I have the same problem.
You can reproduce using Invenio [maint-2.0 branch](https://github.com/inveniosoftware/invenio/tree/285b951e0455ef10a9b97604eaedd3bf1c274f65) and [devscripts](https://github.com/hachreak/invenio-devscripts/blob/master/invenio2-kickstart) to install a virtual machine.
Follow the instruction [to install it](https://github.com/hachreak/invenio-devscripts/blob/master/invenio2-kickstart#L33) and the start tests:

```bash
pip install futures
git checkout maint-2.0
find . -name "*.pyc" -exec rm '{}' \;
pip install -e .[docs] --upgrade
inveniomanage database recreate --yes-i-know
python setup.py test
```

This is the error:
```python
Error in atexit._run_exitfuncs:
Traceback (most recent call last):
  File "/usr/lib/python2.7/atexit.py", line 24, in _run_exitfuncs
    func(*targs, **kargs)
  File "/home/vagrant/.virtualenvs/invenio2/local/lib/python2.7/site-packages/concurrent/futures/process.py", line 76, in _python_exit
    items = list(_threads_queues.items())
AttributeError: 'NoneType' object has no attribute 'items'
Error in atexit._run_exitfuncs:
Traceback (most recent call last):
  File "/usr/lib/python2.7/atexit.py", line 24, in _run_exitfuncs
    func(*targs, **kargs)
  File "/home/vagrant/.virtualenvs/invenio2/local/lib/python2.7/site-packages/concurrent/futures/thread.py", line 35, in _python_exit
    items = list(_threads_queues.items())
AttributeError: 'NoneType' object has no attribute 'items'
Error in sys.exitfunc:
Traceback (most recent call last):
  File "/usr/lib/python2.7/atexit.py", line 24, in _run_exitfuncs
    func(*targs, **kargs)
  File "/home/vagrant/.virtualenvs/invenio2/local/lib/python2.7/site-packages/concurrent/futures/thread.py", line 35, in _python_exit
    items = list(_threads_queues.items())
AttributeError: 'NoneType' object has no attribute 'items'
```

**Comment by agronholm** [comment]
I've committed a changeset that should eliminate these errors. Please try the latest version from master and see if it works for you.

**Comment by agronholm** [comment]
Fixed in 6532a744.
